src/wallet/wallet.cpp

```diff
@@ -3037,6 +3039,7 @@ bool CWallet::SelectCoinsByDenominations(int nDenom, CAmount nValueMin, CAmount
     {
         // masternode-like input should not be selected by AvailableCoins now anyway
         //if(out.tx->vout[out.i].nValue == 1000*COIN) continue;
+        if(fNoDuplicateTxIds && setRecentTxIds.find(out.tx->GetHash()) != setRecentTxIds.end()) continue;
```

Comment:
But will not this lead to the situation that most mixing transactions will contain only 1 input from each participant?

Reply:
No, it shouldn't. Even if it does that's fine for privacy.

Reply:
Yes, it's very fine for privacy. But mixing time will increase significantly.

Reply:
This sounds like a valid concern. I'll have a look.

Reply:
Fees would go up, but I see no situation where a participant would only be able to submit one denom, maybe the first mix when they are all fresh out of create denoms

Reply:
@PastaPastaPasta not only after create denoms. 

What we have: [in most cases](https://github.com/dashpay/dash/blob/e0b0f4e1d9a67dc467734c0adadbfa795534934e/src/privatesend-client.cpp#L1183-L1194) (`nRoundStart = privateSendClient.nPrivateSendRounds`) mixing function will select inputs with highest round originating only from different Txs.

After "create denoms" the participant will be able to provide from 1 to several inputs (depending on amount he is trying to mix - can be several "create denoms" Txs).

And after every mixing round where participant will be able to provide more than 1 input, on the next round in most cases he will not be able to provide more than 1 input because all inputs with highest round will be from the same Tx.

The number of inputs in this situation will always tend to one.